app/models/user.rb

```diff
@@ -0,0 +1,45 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :bigint(8)        not null, primary key
+#  provider               :string           default("email"), not null
+#  uid                    :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  reset_password_token   :string
+#  reset_password_sent_at :datetime
+#  allow_password_change  :boolean          default(FALSE)
+#  remember_created_at    :datetime
+#  sign_in_count          :integer          default(0), not null
+#  current_sign_in_at     :datetime
+#  last_sign_in_at        :datetime
+#  current_sign_in_ip     :string
+#  last_sign_in_ip        :string
+#  confirmation_token     :string
+#  confirmed_at           :datetime
+#  confirmation_sent_at   :datetime
+#  unconfirmed_email      :string
+#  name                   :string
+#  nickname               :string           not null
+#  image                  :string
+#  email                  :string
+#  tokens                 :json
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#
+
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable
+  include DeviseTokenAuth::Concerns::User
+
+  mount_base64_uploader :image, ImageUploader
+  validates :nickname, presence: true, uniqueness: true
+  validates :password, length: { in: 8..20 }, unless: :blank_password?
+
+  def blank_password?
```

Comment:
Since this method is just used internally you should make this private. Just add the word private and every single method below that would be private